src/python/dxpy/__init__.py

```diff
@@ -609,6 +609,16 @@ def ensure_ascii(i):
 
                 response = pool_manager.request(_method, encoded_url, headers=_headers, body=body,
                                                 timeout=timeout, retries=False, **kwargs)
+                # Handle redirection manually for symlink files
+                if response.status in range(300, 399):
+                    url = response.headers.get('Location')
+                    if url:
+                        if try_index > 0:
+                            logger.info("[%s] %s %s: Recovered after %d retries", time.ctime(), method, _url, try_index)
+                        # Make a new request to the URL specified in the Location header
+                        return DXHTTPRequest(url, body, method, headers, auth, timeout, use_compression, jsonify_data, want_full_response,
```

Comment:
sorry but could you please rewrite to some variable, break this `while` loop and fire the new request once the whole block is done? I've realized that any exception thrown by this new call would trigger exception handling of the original request, which may lead to retry of everything and potentially some weird behaviour...